arkouda/random/_generator.py

```diff
@@ -213,6 +213,59 @@ def standard_exponential(self, size=None, method="zig"):
         self._state += size if method.upper() == "INV" else 1
         return create_pdarray(rep_msg)
 
+    def logistic(self, loc=0.0, scale=1.0, size=None):
+        if size is None:
+            # delegate to numpy when return size is 1
+            return self._np_generator.logistic(loc=loc, scale=scale, size=size)
+
+        mu = loc
+        if _val_isinstance_of_union(mu, numeric_scalars):
+            is_single_mu = True
+            if not _val_isinstance_of_union(mu, float_scalars):
+                mu = float(mu)
+        elif isinstance(mu, pdarray):
+            is_single_mu = False
+            if size != mu.size:
+                raise TypeError("array of locs must have same size as return size")
+            if mu.dtype != akfloat64:
+                from arkouda.numeric import cast as akcast
+
+                mu = akcast(mu, akfloat64)
+        else:
+            raise TypeError("logistic only accepts a pdarray or float scalar for loc")
+
+        if _val_isinstance_of_union(scale, numeric_scalars):
+            is_single_scale = True
+            if not _val_isinstance_of_union(scale, float_scalars):
+                scale = float(scale)
+        elif isinstance(scale, pdarray):
+            is_single_scale = False
+            if size != scale.size:
+                raise TypeError("array of scales must have same size as return size")
+            if scale.dtype != akfloat64:
+                from arkouda.numeric import cast as akcast
+
+                scale = akcast(scale, akfloat64)
```

Comment:
Two observations and/or comments: (1) shouldn't loc just be called mu in the first place? (2) since the blocks of code that process mu and scale are identical, this would be a good candidate for moving that code into a function, and calling it for both mu and scale.  This is either (a) a very good comment, or (b) just a case of two people having different programming styles.

Reply:
This is a good comment! 

(1) I kept it `loc` in the first place because that's what the argument is called in the numpy version of `logistic`. The reason I don't keep it `loc` is because I was hitting name collisions in the chapel code with  variable for a locale. I figured `mu` was a good alternative since that's the symbol that loc represents in the formula

(2) I agree I think pulling this out as a helper is a good idea. It's definitely going to come up again when I add another distribution so I'll make that change then